src/core/event/event-bus.js

```diff
@@ -92,7 +92,11 @@ class EventBus extends EventEmitter {
      */
     emitEvent (event, params, cb) {
         var self = this;
-        self.emit(event, params, cb);
+        if (typeof cb === 'function') {
+            self.emit(event, params, cb);
+        } else {
+            console.log('cb passed to emitEvent not a function');
+        }
```

Comment:
I somewhat agree with what you're doing here but I don't think we need it because 1) some events don't expect a callback and 2) we can most likely trust our devs to pass cb correctly (it'll error out anyway if it's not a function and is invoked like one)